kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternFieldsController.java

```diff
@@ -413,12 +417,20 @@ private void generateMeaningForm() {
 
     @FXML
     private void clearView(ActionEvent actionEvent) {
+
         patternFieldsViewModel.setPropertyValue(DISPLAY_NAME, "");
+        patternFieldsViewModel.setPropertyValue(DATA_TYPE, null);
+        removePurpose();
+        removeMeaning();
         patternFieldsViewModel.save(true);
     }
 
     @FXML
     private void onCancel(ActionEvent actionEvent) {
+        //publish close env
+        EvtBusFactory.getDefaultEvtBus().publish(patternFieldsViewModel.getPropertyValue(PATTERN_TOPIC),
+                new PatternPropertyPanelEvent(actionEvent.getSource(), CLOSE_PANEL));
+        clearView(actionEvent);
```

Comment:
is cancel supposed to call clearView? which in turn removes the purpose and meaning?  I thought Sundeep mentioned that cancel isn't supposed to clear... is that right?

Reply:
As per my understanding, Cancel and Done both should clearView. Based on what I was told, every time a user clicks on the pencil icon a clear form should be presented.

Reply:
you are correct, will approve and merge